packages/financial-templates-lib/src/price-feed/ExpressionPriceFeed.js

```diff
@@ -0,0 +1,117 @@
+const assert = require("assert");
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const Web3 = require("web3");
+const { create, all } = require("mathjs");
+const math = create(all, {
+  number: "BigNumber",
+  precision: 100
+});
+
+// Gets balancer spot and historical prices. This price feed assumes that it is returning
+// prices as 18 decimals of precision, so it will scale up the pool's price as reported by Balancer contracts
+// if the user specifies that the Balancer contract is returning non-18 decimal precision prices.
+class ExpressionPriceFeed extends PriceFeedInterface {
+  constructor(priceFeedMap, expression) {
+    super();
+    this.expressionCode = math.parse(expression).compile();
```

Comment:
Will this throw if `expression` is invalid? I think thats a good outcome but just want to make sure

Reply:
It depends on _how_ it is invalid. It will throw if the expression is unparsable, but it will _not_ throw if it's parsable by the engine, but doesn't resolve to a number or if one of the symbols isn't resolved by the scope.

Reply:
Just to follow-up here. It's difficult to guarantee valid-ness of the expression without passing in values for all of the symbols, but we also can't assume which symbol values would be valid in the expression. For instance, passing in 1 for all values would cause the following to throw:
```
(A + B) / (A - B)
```

Reply:
Technically, we could check that all symbols are defined in the map, but I don't think we could check for all types of expression errors.